CHANGELOG.md

```diff
@@ -10,6 +10,12 @@ Note that Sockeye has checks in place to not translate with an old model that wa
 
 Each version section may have have subsections for: _Added_, _Changed_, _Removed_, _Deprecated_, and _Fixed_.
 
+## [1.18.34]
+### Added
+- Constrains input from plain file. 
```

Comment:
This description isn't clear to me: what constrains the input?

Reply:
I can update that description. The only way to input constraints was via json file. 

Now you can use a plain file. In each line, you have a list of space-separated strings (constraints) which correspond to an image (or source sentence). This is a parallel file to the one specified in --input

Reply:
so are these then only single-word constraints? How about several multi-word constraints (which is supported by JSON)?

Reply:
Oh I see. How about having them separated by some special character (e.g., comma or tab)?

Reply:
E.g.: blue sky, cow, grass, ...

Reply:
Then we would need to rewrite the entire parsing logic of separating tokens. You can see how complicated this already is with the factored strings. In my opinion, inputs with constraints and/or factors is inherently structured data and not just plain strings. JSON provides a good format for such use cases, whereas plain file parsing is quite error-prone. 

I don't want to enforce my opinion on this though. Maybe others can chime in on this discussion? We could also separate the constrained decoding for image captioning feature from this API change to unblock this PR.

Reply:
You mean "source sentence", right?

Creating different constraints for an input sentence is already easy to do. You can use the sockeye.lexical_constraints CLI to produce inputs with arbitrary constraints and pass/pipe them directly to sockeye.translate / sockeye.captioner with the `--json-input` flag.

I think we should hold off on changing the overall API / behavior of the API/translate CLI with this PR. This PR should concentrate on enabling constraints for image captioning.

Reply:
Yes. I meant source sentence.

lexical_constraints expects a sentence as input + constraints. Would it work with the image relative path?

I wanted to avoid to use it, because in my case I have to first generate the constraints, then the file with source sentence (image path) + constraints, then apply lexical_constraints and now I can perform inference. It is a bit tricky sequence of steps.

I can revert the changes and remove the option ``--input-constraints``, even though I think it might be useful.

Reply:
I don't know how your captioner currently handles the content of the 'text' field in the JSON object. If it uses it as a relative image path this should already work.

In the end its up to you how you define the API for the image captioning use case, but I vote for keeping the general translation API (sockeye.translate) as is by reverting the changes in the files out side of `sockeye/image_captioning/`.